auth/src/api/services/user-service.ts

```diff
@@ -63,17 +64,17 @@ export class UserService implements UserHandler {
         const userObj = await this.getUserCollectionFromDatastore('username', username);
         Logger.debug('User object from datastore: ', userObj);
         if (Object.keys(userObj).length > 1) {
-            throw new Error('Multiple users with same credentials!');
+            throw new AuthenticationException('Multiple users with same credentials!');
         }
         const user: User = new User(userObj[Object.keys(userObj)[0]]);
         if (!user.isExisting()) {
-            throw new Error('Username or password is incorrect!');
+            throw new AuthenticationException('Username or password is incorrect.');
         }
         if (!user.is_active) {
```

Comment:
Ah, this one is subtle: Please check first, that the username/password combination is correct. if so, check the active state.

This allows for querying for inactive usernames - just try usernames and see, if you get an isactive-response.

this allows you to combine the statements in the if statement:
```
if (!user.isExisting() || !this.isPasswordCorrect(password, user.password)
```